File: widget/src/types/hotel.ts
  export interface Hotel {
    id: string
    name: string
    location: string
    rating: number
    starRating: number
    guestRating: number
    guestRatingCount: number
    price: number
    currency: string
    imageUrl: string
    amenities: string[]
    images: string[]
    latitude?: number   // needed for map — backend must add these
    longitude?: number
  }

  export interface SearchHotelResponse {
    hotels: Hotel[]
    meta: {
      destination: string
      checkIn: string
      checkOut: string
      currency: string
    }
  }

File: TravelAgent/Models/Hotel.cs
namespace TravelAgent.Models
{
    public class Hotel
    {
        public string? Id { get; set; }
        public string? Name { get; set; }
        public string? Location { get; set; }
        public double? Rating { get; set; }
        public double? StarRating { get; set; }
        public double? GuestRating { get; set; }
        public int? GuestRatingCount { get; set; }
        public double? Price { get; set; }
        public double? SupplierPrice { get; set; }
        public double? StrikeThroughPrice { get; set; }
        public string? Currency { get; set; }
        public string? ImageUrl { get; set; }
        public List<string>? Amenities { get; set; }
        public List<string>? Images { get; set; }
        public string? HotelCode { get; set; }
        public string? Provider { get; set; }
    }
}
